framework_tvb/tvb/core/neocom/h5.py

```diff
@@ -309,3 +311,52 @@ def gather_all_references_of_view_model(gid, base_dir, ref_files):
             uuid_files.append(h5_file.path)
             gather_all_references_by_index(h5_file, uuid_files)
         ref_files.extend(uuid_files)
+
+
+def get_all_h5_paths():
+    """
+    This method returns a list of all h5 files and it is used in the migration of data from version 4 to 5.
+    The operations inside a certain project are retrieved in numerical order (1, 2, 3 etc.).
+    The H5 files inside an operation are sorted based on their create_date attribute.
+    """
+    h5_files = []
+    projects_folder = FilesHelper.get_projects_folder()
+
+    for project_name in os.listdir(projects_folder):
+        # Getting operation folders inside the current project
+        project_path = os.path.join(projects_folder, project_name)
+        project_operations = os.listdir(project_path)
+        project_operations_base_names = [os.path.basename(op) for op in project_operations]
+
+        # Build a list of the operation numbers and sort them
+        operation_int_names = []
```

Comment:
This way of iterating through the projects and operations might not cover all cases. 
It would be better to sort the operations by their creation date and iterate over them accordingly, this is what we do also at import.
In your case, since the operations are already in DB, it might be easier to take the creation date from there, instead of reading it from each Operation.xml.